### PR TITLE
fix: Requests to cancel job no longer error due to missing target.

### LINKED ISSUE #442: Requests to cancel jobs fail due to missing targets
The proto spec for the cancel jobs request was missing a required `target` field. Without this, the request always fails. A recent update to the gRPC client adds this missing field, we need to pull that update in and add the field.

### PR DESCRIPTION
**Pull request by MarquessV**
closes #442

### CODE REVIEW

File: crates/lib/src/qpu/api.rs [hunk 1]
```diff
@@ -131,9 +131,8 @@ pub async fn submit(
 /// This action is *not* atomic, and will attempt to cancel every job even when some jobs cannot be
 /// cancelled. A job can be cancelled only if it has not yet started executing.
 ///
-/// Success response indicates only that the request was received. Cancellation is not guaranteed,
-/// as it is based on job state at the time of cancellation, and is completed on a best effort
-/// basis.
+/// Cancellation is not guaranteed, as it is based on job state at the time of cancellation, and is
+/// completed on a best effort basis.
```
Comment (Shadow53) [inline crates/lib/src/qpu/api.rs:135]: Should be explicit about whether an error is returned if a job cannot be cancelled. As a user, I would also want to know what would happen if I passed in two jobs, where one is already executing but the other can be cancelled.

Comment (Shadow53) [reply crates/lib/src/qpu/api.rs:135]: FWIW the idiomatic thing to do is have a
```rust
/// # Errors
```
section in the comment

Comment (Shadow53) [reply crates/lib/src/qpu/api.rs:135]: May also replace some of this comment with a "See `cancel_job`"

**Review by Shadow53 — APPROVED** [review]
Some notes on doc comments, but the code looks good

File: crates/lib/src/qpu/api.rs (hunk 1 above)
Comment (MarquessV) [reply crates/lib/src/qpu/api.rs:135]: Added a note on what errors to expect when job cancellation fails.

**Pull request merged**